hub/core/storage/s3.py

```diff
@@ -125,9 +160,23 @@ def __delitem__(self, path):
         """
         self.check_readonly()
         self._check_update_creds()
+        path = posixpath.join(self.path, path)
         try:
-            path = posixpath.join(self.path, path)
             self.client.delete_object(Bucket=self.bucket, Key=path)
+        # catch expired token error
+        except botocore.exceptions.ClientError as err:
+            if (
+                err.response["Error"]["Code"] == "ExpiredToken"
+                and self.loaded_creds_from_environment
+            ):
+                self._locate_and_load_creds()
```

Comment:
Modularize lines similar to 172-177 maybe using a context manager.

Check if the outer block can also be modularized.